ext/hal/nxp/imx/README

```diff
@@ -0,0 +1,41 @@
+iMX7D Port
```

Comment:
The drivers in this HAL will be used also for other SoC's (I am going to add iMX6SoloX's M4 core / Udoo Neo board in the near future). Therefore the iMX7D mentions should be changed to something more generic. Or I can change it once I add the other SoC support.

FYI, the package FreeRTOS_BSP_1.0.1_iMX6SX contains the following files in the platform/drivers folder, which are not in FreeRTOS_BSP_1.0.1_iMX7D:
- adc_imx6sx.h
- adc_imx6sx.c
- ccm_analog_imx6sx.h
- ccm_analog_imx6sx.c
- ccm_imx6sx.h
- ccm_imx6sx.c
- rdc_defs_imx6sx.h
- epit.h
- epit.c

FreeRTOS_BSP_1.0.1_iMX7D has the following specific files:
- adc_imx7d.h
- adc_imx7d.c
- ccm_analog_imx7d.h
- ccm_analog_imx7d.c
- ccm_imx7d.h
- ccm_imx7d.c
- rdc_defs_imx7d.h
- gpt.h
- gpt.c

Other differences in the platform/drivers folder are just some typos in the code comments.

I will add the missing files from FreeRTOS_BSP_1.0.1_iMX6SX as part of my future pull request, I think it should be placed in the same folder. But it will have multiple source URLs etc.

Reply:
@stanislav-poboril,

> The drivers in this HAL will be used also for other SoC's (I am going to add iMX6SoloX's M4 core / Udoo Neo board in the near future). Therefore the iMX7D mentions should be changed to something more generic. Or I can change it once I add the other SoC support.

This is a great news. But as soon as I don't have an iMX6SoloX board I can't validate in this hardware and rename the files and configuration names to be more generic. I think that the best way of doing this is when you submit your changes you can make the appropriated generalizations.

Reply:
Ok, I will do it along with my changes.

Reply:
> The drivers in this HAL will be used also for other SoC's (I am going to add iMX6SoloX's M4 core / Udoo Neo board in the near future). Therefore the iMX7D mentions should be changed to something more generic. Or I can change it once I add the other SoC support.

@stanislav-poboril do you have something running on the Udoo Neo?  I was going to look at doing this next week, but if you've already got something I'll hold off.

Reply:
@galak, I have hello_world running. I would like to rebase it against this pull request since it shares part of HAL. Then I would like to create a shim for Messaging Unit driver.